add islower/isupper functions

stevengj opened the pull request:
Closes #195.

@maartenbreddels, this adds `islower` and `isupper` functions that should correspond to the `Lowercase` and `Uppercase` derived properties.   I figured out a trick to do it without expanding the data tables, just slightly tweaking the table entries of lower/uppercase characters that don't yet have upper/lowercase mappings, respectively.

~~I still need to add a test.~~

maartenbreddels commented:
Very cool, if you give me some time, I could test this with https://github.com/apache/arrow/pull/7656

stevengj commented:
Added some tests and fixed a bug.  This seems to be working (it returns true for `isupper`/`islower` if and only if the character is listed as such in `DerivedProperties.txt`).
